src/date.c

```diff
@@ -1361,8 +1471,13 @@ static void strftimeFunc(
       }
       case 's': {
         if( x.useSubsec ){
+#if defined(SQLITE_OMIT_FLOATING_POINT)
+          sqlite3_str_appendf(&sRes,"%.3f",
```

Comment:
If it doesn't use floating point numbers how ca it format 3 decimals?

Reply:
I'm inclined to think that not all floating point operations require non-general registers. In September, using datatype `float` would work without an error, even though it is a floating point operation. This is similar to why we could conduct multiplications in the module without `-mgeneral-regs-only` bugging. It is just some specific operations that bug and require editing.

Reply:
For reference, I think this can pass because the initial value is actually an integer. With SQLITE_OMIT_FLOATING_POINT, x.iJD is `long`, and the other numbers printed in `(x.iJD - 21086676*(i64)10000000)/1000)` are also integers. So we're not actually printing anything with double floating point, and the result is going to be something ending in .000. We could change it to %d as well, as .000 denotes the milliseconds and will always be .000 in our implementation. I have no problem to change it if we consider it more appropriate.